File: packages/typescript/src/roles/project-manager.ts
import { Agent } from "../agent.js";
import { AgentRole, TaskType } from "../types.js";
import type { AgentConfig } from "../types.js";

const HANDLED_TASK_TYPES: TaskType[] = [
  TaskType.SprintPlanning,
  TaskType.Documentation,
];

export class ProjectManager extends Agent {
  constructor(config: Partial<AgentConfig> = {}) {
    super({
      role: AgentRole.ProjectManager,
      ...config,
    });
  }

  canHandle(taskType: TaskType): boolean {
    return HANDLED_TASK_TYPES.includes(taskType);
  }
}

File: packages/python/src/tcs_bpi_ai_team/roles/project_manager.py
"""Project Manager agent."""

from __future__ import annotations

from tcs_bpi_ai_team.agent import Agent
from tcs_bpi_ai_team.types import AgentConfig, AgentRole, TaskType

_HANDLED_TASK_TYPES = [
    TaskType.SPRINT_PLANNING,
    TaskType.DOCUMENTATION,
]


class ProjectManager(Agent):
    """Project Manager — task creation, sprint planning, status tracking."""

    def __init__(self, config: AgentConfig | None = None) -> None:
        super().__init__(config or AgentConfig(role=AgentRole.PROJECT_MANAGER))

    def can_handle(self, task_type: TaskType) -> bool:
        return task_type in _HANDLED_TASK_TYPES
